### PR TITLE
Nfs provisioner support for ppc64le + NFS server and path variables

### PR DESCRIPTION
**Pull request by schabrolles**
- NFS provisioner support for ppc64le (quay.io image is not multiarch, need to use another image)

- add the possibility to specify another nfs server and path by using two variables (`nfs_server` and `nfs_path`) ... but keep default value to be `helper.ipaddr` and `/export`

### CODE REVIEW

**Comment by christianh814** [comment]
@schabrolles Where would someone define `nfs_server` and `nfs_path`? In the `vars.yaml`?

File: templates/nfs-provisioner-deployment.yaml.j2 [hunk 1]
```diff
@@ -17,19 +17,23 @@ spec:
...
+{% endif %}
           volumeMounts:
             - name: nfs-client-root
               mountPath: /persistentvolumes
           env:
             - name: PROVISIONER_NAME
               value: nfs-storage
             - name: NFS_SERVER
-              value: {{ helper.ipaddr }}
+              value: {{ nfs_server | default(helper.ipaddr) }}
```
Comment (bpradipt) [inline templates/nfs-provisioner-deployment.yaml.j2:32]: IMHO this can be in a separate patch to add support for external NFS server and separate from using the right NFS client provisioner image per architecture. 
@christianh814 what do you think ?

File: templates/nfs-provisioner-deployment.yaml.j2 [hunk 2]
```diff
@@ -17,19 +17,23 @@ spec:
       serviceAccountName: nfs-client-provisioner
       containers:
         - name: nfs-client-provisioner
+{% if ansible_architecture == "ppc64le" %}
+          image: ibmcom/nfs-client-provisioner-ppc64le:latest
```
Comment (bpradipt) [inline templates/nfs-provisioner-deployment.yaml.j2:21]: How about creating a variable for nfs-client-provisioner image and use separate values for each architecture ?
The vars-ppc64le.yaml will have the ppc64le specific image and the default will be the quay image.

File: templates/nfs-provisioner-deployment.yaml.j2 (hunk 1 above)
Comment (christianh814) [reply templates/nfs-provisioner-deployment.yaml.j2:32]: So in general, I like this approach because it keeps the default behaviour (which what I strive for) but gives the ability to extend it, if need be.

In general, as long as the default behaviour is preserved, I'm no too picky on the implementation.

**Comment by schabrolles** [comment]
> @schabrolles Where would someone define `nfs_server` and `nfs_path`? In the `vars.yaml`?

It could be in vars.yaml, or in the ansible command `-e nfs_server=10.1.1.1 -e nfs_path=/nfsdata`

The main idea is to keep the default but let the possibility to the user to overwrite them if needed.

File: templates/nfs-provisioner-deployment.yaml.j2 (hunk 2 above)
Comment (christianh814) [reply templates/nfs-provisioner-deployment.yaml.j2:21]: @bpradipt So I like this, but would still like to provide a default if none is provided.

ppc64le workflow:

1) If the user provides a custom Image, then that specific image is used.
2) If the user does NOT provide the image, it defaults to `ibmcom/nfs-client-provisioner-ppc64le:latest`

x86_64 workflow (similar):

1) If the user provides a custom Image, then that specific image is used.
2) If the user does NOT provide the image, it defaults to `quay.io/external_storage/nfs-client-provisioner:latest`

So in general I like the idea of the user being able to specify an image, but I would also like to have a "reasonable default" if the user doesn't provide it

**Comment by christianh814** [comment]
> The main idea is to keep the default but let the possibility to the user to overwrite them if needed.

I like this approach! This is good for a first pass and I'll run it though my testing scripts later today

**Comment by christianh814** [comment]
@schabrolles @bpradipt I'm going to test this (and hopefully merge it to `devel`) today.

I think this is a good first pass, and we'll make updates if we need to

**Comment by schabrolles** [comment]
Ok ... We may need to update the documentation somewhere about the 2 new variables `nfs_server`and `nfs_path`... but I don’t know where to put this ... Any suggestions?

File: templates/nfs-provisioner-deployment.yaml.j2 (hunk 2 above)
Comment (christianh814) [reply templates/nfs-provisioner-deployment.yaml.j2:21]: Coming back to this, I've made some modifications to this file and I think I know what you're talking about bpradipt. The ability to specify image.

Right now you can do..

```
nfs:
  server: 192.168.7.77
  path: /export
```

I could see in the future doing


```
nfs:
  server: 192.168.7.77
  path: /export
  image: quay.io/external_storage/nfs-client-provisioner:latest
```

However, since this specifically uses the nfs-provisioner from the [kuberentes upstream incubator](https://github.com/kubernetes-incubator/external-storage/blob/master/nfs-client/README.md), other than ppc64le, when would someone provide a custom image for this?

In general I'm for having the ability to specify an image (even if it's a "hidden" feature)

**Comment by christianh814** [comment]
Quay.io was down yesterday so I'll continue testing today
